picard/tagger.py

```diff
@@ -517,75 +517,54 @@ def add_files(self, filenames, target=None):
             for file in new_files:
                 file.load(partial(self._file_loaded, target=target))
 
+    def _scan_dir(self, ignore_hidden, folders, recursive):
+        files = []
+        local_folders = list(folders)
+        while len(local_folders) > 0:
+            current_folder = local_folders.pop(0)
+            current_folder_files = []
+            for entry in os.scandir(current_folder):
+                if ignore_hidden and is_hidden(entry.path):
+                    continue
+                if entry.is_dir() and recursive:
+                    local_folders.extend([entry.path])
+                else:
+                    current_folder_files.extend([entry.path])
+
+            number_of_files = len(current_folder_files)
+            if number_of_files:
+                mparms = {
+                    'count': number_of_files,
+                    'directory': current_folder,
+                }
+                log.debug("Adding %(count)d files from '%(directory)r'" %
+                          mparms)
+                from gettext import ngettext
+                self.window.set_statusbar_message(
+                    ngettext(
+                        "Adding %(count)d file from '%(directory)s' ...",
+                        "Adding %(count)d files from '%(directory)s' ...",
+                        number_of_files),
+                    mparms,
+                    translate=None,
+                    echo=None
+                )
+                files.extend(current_folder_files)
+        return files
+
     def add_directory(self, path):
-        if config.setting['recursively_add_files']:
-            self._add_directory_recursive(path)
-        else:
-            self._add_directory_non_recursive(path)
+        from threading import Thread
```

Comment:
We usually place all imports at the top. 

Just out of interest: Is there any direct advantage of using `threading.Thread` here over the existing QThread pool? Or is this more about starting to move away from QThreads?

Reply:
There shouldn't be any difference in this case, but the entire threading setup is pretty confusing to me. 
I can move the import to the top.